pkg/ipam/ipam.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2017-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/cilium/cilium/pkg/datapath"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/ip"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	"k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "ipam")
+)
+
+type ErrAllocation error
+
+// Family is the type describing all address families support by the IP
+// allocation manager
+type Family string
+
+const (
+	IPv6 Family = "ipv6"
+	IPv4 Family = "ipv4"
```

Comment:
These appear to duplicate `familyIPv6` and `familyIPv4`.

Reply:
Yes, I did not want to export untyped constant and I didn't want to refactor all code either and keep the changes small.